src/SSHLibrary/abstractclient.py

```diff
@@ -202,7 +202,9 @@ def login(self, username=None, password=None, allow_agent=False, look_for_keys=F
         """
         keep_alive_interval = int(TimeEntry(keep_alive_interval).value)
         username = self._encode(username)
-        if not password and not allow_agent:
+        if not password:
+            password = None
+        if password and not allow_agent:
```

Comment:
This is obviously correct, Commit 355ff5ca7db2 changed `password != None` to `not password`, which is the opposite